src/engine/strat_engine/thinpool.rs

```diff
@@ -165,11 +166,12 @@ impl ThinPool {
     }
 
     /// The status of the thin pool as calculated by DM.
-    pub fn check(&self, dm: &DM) -> EngineResult<ThinPoolStatus> {
+    pub fn check(&mut self, dm: &DM) -> EngineResult<ThinPoolStatus> {
         let thinpool = try!(self.thin_pool.status(dm));
         try!(self.mdv.check());
 
         let filesystems: Vec<_> = try!(self.filesystems
+                                           .borrow_mut()
```

Comment:
This is odd... this really should be a `self.filesystems.iter()` and I know that doesn't work right now, but it should be fixed. Opened #492.